### PR TITLE
Convert the chef-bcpc::system recipe to an Ansible task

### PR DESCRIPTION
**Pull request by tj90241**
Convert the chef-bcpc::system recipe to an Ansible task

### CODE REVIEW

File: ansible/playbooks/roles/common/defaults/main/base.yml [hunk 1]
```diff
@@ -244,3 +244,34 @@ ubuntu_apt:
...
+hardware:
+  # Select desired I/O scheduler to be applied at startup (deadline, noop, cfq)
+  io_scheduler: deadline
+
+kernel:
+  # Pin the system kernel to a fixed version?
+  pin_version: false
+  version: ''
+
+system:
```
Comment (justinjpacheco) [inline ansible/playbooks/roles/common/defaults/main/base.yml:266]:
```suggestion
system_parameters_kernel_pid_max: 4194303
system_parameters_net.nf_conntrack_max: 262144
```

Comment (justinjpacheco) [reply ansible/playbooks/roles/common/defaults/main/base.yml:266]: perhaps we should explore building our ansible attributes in a much flatter fashion?

File: ansible/playbooks/roles/common/templates/grub/default.j2 [hunk 2]
```diff
@@ -1,24 +1,24 @@
...
+{% endif -%}
 GRUB_HIDDEN_TIMEOUT=0
 GRUB_HIDDEN_TIMEOUT_QUIET=true
 GRUB_TIMEOUT=0
 GRUB_DISTRIBUTOR=`lsb_release -i -s 2> /dev/null || echo Debian`
 GRUB_CMDLINE_LINUX_DEFAULT="vga=normal nomodeset 3 console=tty0 console=tty1"
-# use <%= @node['bcpc']['hardware']['io_scheduler'] %> I/O scheduler
-GRUB_CMDLINE_LINUX="<%= @cmdline %>"
+# use {{ hardware['io_scheduler'] }} I/O scheduler
+GRUB_CMDLINE_LINUX="{{ cmdline }}"
```
Comment (justinjpacheco) [inline ansible/playbooks/roles/common/templates/grub/default.j2:21]: if cmdline is an array, won't it need to be piped into join?

Comment (tj90241) [reply ansible/playbooks/roles/common/templates/grub/default.j2:21]: Did the processing for this in the yml - can move it here if you think it makes more sense.
`cmdline: "{{ ' '.join(['elevator=' + hardware['io_scheduler']] + grub['cmdline_linux']) }}"`

File: ansible/playbooks/roles/common/defaults/main/base.yml (hunk 1 above)
Comment (tj90241) [reply ansible/playbooks/roles/common/defaults/main/base.yml]: Done, just have to make the jinja a little less unwiedly.

**Comment by tj90241** [comment]
Folded into https://github.com/bloomberg/chef-bcpc/pull/1695

**Pull request closed without merging**